### ISSUE TITLE
Python3 compatibility issue

### ISSUE THREAD

**Issue by LarsMichelsen** [fschulze/pytest-flakes#9]
I am trying to make some library of mine Python2/3 compatbile. For testing I use pytest with your flakes extension. Thanks for this.

With Python 2.7 the test run fine:

```
# python2 setup.py test
running pytest
running egg_info
writing pmatic.egg-info/PKG-INFO
writing top-level names to pmatic.egg-info/top_level.txt
writing dependency_links to pmatic.egg-info/dependency_links.txt
reading manifest file 'pmatic.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'pmatic.egg-info/SOURCES.txt'
running build_ext
================================================================= test session starts =================================================================
platform linux2 -- Python 2.7.9 -- py-1.4.25 -- pytest-2.6.3
plugins: cache, flakes
collected 34 items

setup.py s
update-file-headers.py s
examples/list_available_methods.py s
examples/list_rooms.py s
examples/list_shutter_contact_states.py s
examples/switch_off_all_lamps.py s
examples/switch_on_single_power_switch.py s
examples/lowlevel_api/list_shutter_contact_states.py s
pmatic/__init__.py s
pmatic/entities.py s
pmatic/utils.py s
pmatic/api/__init__.py s
pmatic/api/abstract.py s
pmatic/api/local.py s
pmatic/api/remote.py s
tests/TestAPI.py s
tests/test_api.py s......
tests/test_entities.py s.....
tests/test_utils.py s....

================================================== 15 passed, 19 skipped, 1 warnings in 0.11 seconds =================================================
```

But with python3 there is an exception within pytest_flakes.py:

```
# python3 setup.py test
running pytest
running egg_info
writing pmatic.egg-info/PKG-INFO
writing top-level names to pmatic.egg-info/top_level.txt
writing dependency_links to pmatic.egg-info/dependency_links.txt
reading manifest file 'pmatic.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'pmatic.egg-info/SOURCES.txt'
running build_ext
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/usr/lib/python3/dist-packages/_pytest/main.py", line 80, in wrap_session
INTERNALERROR>     config.do_configure()
INTERNALERROR>   File "/usr/lib/python3/dist-packages/_pytest/config.py", line 618, in do_configure
INTERNALERROR>     self.hook.pytest_configure(config=self)
INTERNALERROR>   File "/usr/lib/python3/dist-packages/_pytest/core.py", line 413, in __call__
INTERNALERROR>     return self._docall(methods, kwargs)
INTERNALERROR>   File "/usr/lib/python3/dist-packages/_pytest/core.py", line 424, in _docall
INTERNALERROR>     res = mc.execute()
INTERNALERROR>   File "/usr/lib/python3/dist-packages/_pytest/core.py", line 315, in execute
INTERNALERROR>     res = method(**kwargs)
INTERNALERROR>   File "/root/git/pmatic/pytest_flakes-1.0.1-py3.4.egg/pytest_flakes.py", line 36, in pytest_configure
INTERNALERROR>     config._flakes = FlakesPlugin(config)
INTERNALERROR>   File "/root/git/pmatic/pytest_flakes-1.0.1-py3.4.egg/pytest_flakes.py", line 43, in __init__
INTERNALERROR>     self.mtimes = config.cache.get(HISTKEY, {})
INTERNALERROR> AttributeError: 'Config' object has no attribute 'cache'
```

I added these attributes to my setup.py:

```
    setup_requires=['pytest-runner'],
    tests_require=['pytest', 'pytest-flakes'],
)
```

And this is my pytest.ini:

```
[pytest]
testpaths = tests
addopts = --flakes
flakes-ignore =
    playground.py ALL
```

While collecting the information together I found a solution for the issue. When adding pytest-cache to the setup.py tests_require argument like this it works:

```
    tests_require=['pytest', 'pytest-flakes', 'pytest-cache'],
```

**Comment by fschulze** [comment]
AFAIK pytest-cache was included into the latest pytest releases. Make sure you are using the current version.

**Comment by LaMi-** [comment]
I tested python 2 and 3 with pytest 2.6.3 from official Debian 8 repos:

```
# dpkg -l | egrep python.*-pytest
ii  python-pytest                  2.6.3-2                      all          Simple, powerful testing in Python
ii  python3-pytest                 2.6.3-2                      all          Simple, powerful testing in Python3
```

The strange thing is that 2 and 3 seem to have different handling here.

**Comment by fschulze** [comment]
pytest-cache was included into pytest with 2.8.0, so if you use an older version you just have to add the pytest-cache dependency manually. That it's different between Python 2 and 3 is weird, but maybe pytest-cache is installed as a global package for 2 on your system?

**Comment by fschulze** [comment]
Maybe you could write a blurp for the readme mentioning the error and the solution for pre 2.8.0 pytest? I'd be happy to merge a PR for that.
